### PR TITLE
Homoglyphs check

### PR DESCRIPTION
**Pull request by KarenAJ**

### CODE REVIEW

File: app/core/management/utils/xis_internal.py [hunk 1]
```diff
@@ -209,3 +217,22 @@ def bleach_data_to_json(rdata):
             rdata[key] = bleach_data_to_json(rdata[key])
 
     return rdata
+
+
+def confusable_homoglyphs_check(id_num, data, data_is_safe):
```
Comment (JDTobin) [inline app/core/management/utils/xis_internal.py:222]: Don't have data_is_safe as an argument

File: app/core/management/utils/xis_internal.py [hunk 2]
```diff
@@ -209,3 +217,22 @@ def bleach_data_to_json(rdata):
...
+
+        # if string, Check homoglyph
+        if isinstance(data[key], str):
+            if bool(confusables.is_dangerous(data[key])):
+                data_is_safe = False
+                required_recommended_logs(id_num, "homoglyphs", key)
+                logger.error(categories.unique_aliases(data[key]))
+        # if dict, enter dict
+        if isinstance(data[key], dict):
+            data_is_safe = confusable_homoglyphs_check(id_num, data[key],
```
Comment (JDTobin) [inline app/core/management/utils/xis_internal.py:235]: check if recursive call returns false, to set data_is_safe to false

File: app/core/management/utils/xis_internal.py [hunk 3]
```diff
@@ -209,3 +217,22 @@ def bleach_data_to_json(rdata):
...
+
+
+def confusable_homoglyphs_check(id_num, data, data_is_safe):
+    """Checks for dangerous homoglyphs."""
+
+    for key in data:
+
+        # if string, Check homoglyph
+        if isinstance(data[key], str):
+            if bool(confusables.is_dangerous(data[key])):
```
Comment (JDTobin) [inline app/core/management/utils/xis_internal.py:229]: Could combine if statements here

File: app/api/serializers.py [hunk 4]
```diff
@@ -121,6 +125,19 @@ def validate(self, data):
         data['record_status'] = record_status_result
         data['date_validated'] = timezone.now()
 
+        # confusable homoglyphs check
+
+        safe_data = confusable_homoglyphs_check(id_num, json_metadata, True)
```
Comment (JDTobin) [inline app/api/serializers.py:130]: Don't pass in True

File: app/api/serializers.py [hunk 5]
```diff
@@ -205,6 +222,17 @@ class Meta:
     def validate(self, data):
         """Assign active status to supplemental data """
 
+        json_metadata = data.get('metadata')
+        id_num = data.get('unique_record_identifier')
+
+        # confusable homoglyphs check
+        safe_data = confusable_homoglyphs_check(id_num, json_metadata, True)
```
Comment (JDTobin) [inline app/api/serializers.py:229]: Don't pass in True

**Review by JDTobin — CHANGES_REQUESTED** [review]
remove data_is_safe argument

**Review by JDTobin — CHANGES_REQUESTED** [review]
remove data_is_safe argument

**Review by sammy-sandhu — APPROVED** [review]

**Review by JDTobin — APPROVED** [review]
LGTM

**Pull request merged**